chat-plugins/info.js

```diff
@@ -2614,18 +2614,33 @@ var commands = exports.commands = {
 		if (!this.can('declare', null, room)) return false;
 		if (!this.canBroadcast()) return;
 		if (this.room.isPersonal && !this.user.can('announce')) {
-			this.errorReply("Images are not allowed in personal rooms.");
-			return false;
+			return this.errorReply("Images are not allowed in personal rooms.");
 		}
 
 		var targets = target.split(',');
-		if (targets.length !== 3) {
-			return this.parse('/help showimage');
-		}
+		if (targets.length !== 3) return this.parse('/help showimage');
+
+		var image = targets[0].trim();
+		if (!image) return this.errorReply('No image URL was provided!');
+		if (!/^https?:\/\//.test(image)) image = '//' + image;
+
+		var width = targets[1].trim();
+		if (!width) return this.errorReply('No width for the image was provided!');
+		if (!isNaN(width)) width += 'px';
+
+		var height = targets[2].trim();
+		if (!height) return this.errorReply('No height for the image was provided!');
+		if (!isNaN(height)) height += 'px';
+
+		image = Tools.escapeHTML(image);
+		width = Tools.escapeHTML(width);
```

Comment:
This isn't enough, you still need to validate that `width` will be interpreted as an actual CSS value. It's also against convention to escape HTML here rather than at the site of where the value becomes HTML (e.g. `width: ' + Tools.escapeHTML(width)`.